### PR TITLE
Issue #14937: Migrated section 5.3 Camel case: defined to follow chapter wise testing

### LINKED ISSUE #15138: Bdd violations messages does not allow skip of leading '.*'
problem is detected at:
https://github.com/checkstyle/checkstyle/pull/15126#discussion_r1654047096

if violation message is long we have to use short/small section of it sometime in the middle to keep most meaning  
this is required now: `// violation above '.* 'newCustomerID' .* no more than '1' .* capital letters.'`
is we try to remove '.*' test is failing

desirable: `// violation above 'newCustomerID.* no more than '1' .* capital letters.'`

### PR DESCRIPTION
**Pull request by Zopsss**
Closes #15138
Part of #14937

### CODE REVIEW

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+            void YouTubeImporter() {}
+
+            void YoutubeImporter() {}
+    };
+}
+
+class AbbreviationsIncorrect {
+
+    int newCustomerID; // violation 'Abbreviation .* 'newCustomerID' .* '1' .*'
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java:51]: too cryptic
https://checkstyle.org/checks/naming/abbreviationaswordinname.html#Violation_Messages we have single message 
`Abbreviation in name ''{0}'' must contain no more than ''{1}'' consecutive capital letters.`

can we do `no more than ''1'' consecutive capital letters.` ?
or `no more than ''1'' .* capital letters.`

**Review by romani — CHANGES_REQUESTED** [review]
item

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java (hunk 1 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: Tried this:

https://github.com/checkstyle/checkstyle/blob/ddd818855aa6427ba31af0dc895c4f260b04bc38/src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java#L78-L82

it is close to cross the 100 width column limit

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            void YouTubeImporter() {}
+
+            void YoutubeImporter() {}
+    };
+}
+
+class AbbreviationsIncorrect {
+
+    int newCustomerID;
+    // violation above '.* 'newCustomerID' .* no more than '1' .* capital letters.'
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java:52]: // violation above ' 'newCustomerID' .* more than '1' .* capital letters.'

**Review by romani — CHANGES_REQUESTED** [review]
This more readable for my mind

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java (hunk 2 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: done

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: @Zopsss , please remove heading `.*`

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: And single quotes also can be removed

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: Ok I see that you tried , but there is problems

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: But we can remove single quotes

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java]: fixed by https://github.com/checkstyle/checkstyle/issues/15138
commit in this PR.

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
Rebased

**Pull request merged**